### ISSUE TITLE
[REVIEW]: Vizumap: an R package for visualising uncertainty in spatial data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2409]
**Submitting author:** @lydialucchesi (<a href="http://orcid.org/0000-0002-1901-4301">Lydia Lucchesi</a>)
**Repository:** <a href="https://github.com/lydialucchesi/Vizumap" target ="_blank">https://github.com/lydialucchesi/Vizumap</a>
**Version:** v1.2.0
**Editor:** @bstabler
**Reviewers:** @nuest, @GISerDaiShaoqing
**Archive:** <a href="https://doi.org/10.5281/zenodo.4554558" target="_blank">10.5281/zenodo.4554558</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645/status.svg)](https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645"><img src="https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645/status.svg)](https://joss.theoj.org/papers/3caaea5389aa90e287f459bf621c8645)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/lydialucchesi/Vizumap" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@nuest & @GISerDaiShaoqing, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @bstabler know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/lydialucchesi/Vizumap">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@lydialucchesi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/lydialucchesi/Vizumap">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@lydialucchesi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @bstabler it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1002/sta4.150 is OK
- 10.1071/MF17237 is OK
- 10.1214/16-AOAS950 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02409/joss.02409/10.21105.joss.02409.pdf)

**Comment by bstabler** [comment]
Hi @lydialucchesi, thanks for the submittal.  I goofed this up since I'm not supposed to assign myself both editor and reviewer.  I'm working on getting us another editor and/or reviewer.  Thanks.

**Comment by bstabler** [comment]
@whedon remove @bstabler as reviewer

**Comment by whedon** [comment]
OK, @bstabler is no longer a reviewer

**Comment by bstabler** [comment]
@whedon add @nuest as reviewer

**Comment by whedon** [comment]
OK, @nuest is now a reviewer

**Comment by bstabler** [comment]
@whedon add @mikejohnson51 as reviewer

**Comment by whedon** [comment]
OK, @mikejohnson51 is now a reviewer

**Comment by nuest** [comment]
Thanks for the chance to contribute to JOSS!

I'll likely give this a go middle of next week.

**Comment by lydialucchesi** [comment]
Hi Ben,

Thanks for the update and finding us several reviewers.

Cheers,
Lydia

**Comment by lydialucchesi** [comment]
Hi Daniel,

Sounds great - thanks for letting me know.

Best,
Lydia

**Comment by bstabler** [comment]
hi @nuest & @mikejohnson51, how's the review coming along?  Thanks!

**Comment by mikejohnson51** [comment]
Hi @bstabler & @lydialucchesi,

I hope to have this done early next week if that is acceptable. Thanks!

**Comment by nuest** [comment]
Will work on it tomorrow - apologies for the delay.

**Comment by nuest** [comment]
## First review notes

### General

- **Authorship**
  - The first author @lydialucchesi has primarily contributed to the JOSS paper - at least judging from the commits. Major code contributions were made by @pkuhnert. The third author has not made any contribution to the software himself, but I assume there is a significant contribution in algorithm design and the idea behind the package. I assume the contributors discussed this, but IMO it would be best to capture this with a short comment by @pkuhnert and Christopher K. Wikle (does not seem to be on GitHub) ? @bstabler what's your take on this?
- **License**
  - The README says "The package Vizumap version 1.1.0 is licensed under GPL-3 (see LICENSE file)", yet the `LICENSE` file contains a "CSIRO Open Source Software License Agreement (GPLv3)", which does not contain the full text of said license, and also references another software (`asremlREplot`) - please clarify in the README, and suggest to include full license text.
  - This license is AFAICT not OSI-approved
  -  I assume there is no way around a non-standard license for you because of funding?
  - The license statement reads strange because it includes the version. Are other versions released under different licenses? The version `1.1.0` also does not match the release version on GitHub (`1.0.0`) nor the version in the `DESCRIPTION` file (`1.2.0`) - please clarify/fix.

### Documentation

- I personally prefer to suggest installation via `remotes`, because it is much quicker to install than `devtools`. Just an idea for the README.
- Both papers referenced in the README are not open access - maybe you can add links to postprints or otherwise deposited copies of the PDFs? I cannot read them.
- The README and papers mention four methods, the vignette mentions only three. Please adjust/clarify.
- The reference in the README uses two different Zenodo DOIs, please fix.
- The commands in the vignette work :+1: There was a slight problem with plotting though on my machine, see sessionInfo output below: The worl map boundaries of Mexico are plottet over Alaska: ![image](https://user-images.githubusercontent.com/1325054/88483177-6ff28500-cf66-11ea-879a-505ec53ef76f.png)
- I get a warning then running the `pixelate` example in the vignette. _Is that expected?_
```r
> pix <- pixelate(ca_geo, id = "region")
Warning message:
In spTransform(shapefile, CRS("+proj=longlat +datum=WGS84")) :
  NULL source CRS comment, falling back to PROJ string
```
- `ca_data$region %in% pix$region` in the vignette would have a shorter output if wrapped with `all(..)`
- Excellent job mentioning which commands might take a bit longer to run in the vignette!
- The vignette assumes readers execute commands from top to bottom, also with the naming of variables. I suggest to use longer variable names for better understandability and readability (not `map` and `map2`)
- **Community guidelines** are missing AFAICS. The `usethis` package can be very helpful to add these.
- **Automated tests** are missing AFAICS. I know tests for visualisations are hard, but you could, and should, have tests for the data preparations functions.

### Paper

- **Statement of need**: could you clarify the targeted users of your package? You mention decision makers as the readers of the created maps, but who creates them? (minor suggestion)
- **State of the field**: please clarify if on other packages (in R) exist.
- "finding methods that add additional elements" - you probably mean more something like "finding methods that can communicate additional elements in an understandable and meaningful way"? You can easily add more elements, but adding userful once seems to be what your methods do.
- "functions can be found inthe package download" - suggest to rephrase to "is available after package installation". It's pretty straightforward these days to get a GitHub pages website for an R package (`pkgdown`, `tic` can help), so if you don't plan to submit the package to CRAN, then having a little website would be great for users. _This is of course unrelated to the JOSS submission_.
- Can you add links to the code that produces the examples in the paper?
- You really have to zoom into the pixel map example to see the pixels. I strongly suggest to show a zoomed in area of the map only, to make the pixel immediately visible for the reader.

### Code

- `DESCRIPTION` file
  - Double field `Maintainer: Maintainer:`
  - `Description` field is very short.
  - `Authors@R` does not match the JOSS paper authors - please clarify.
  - Since the authors are academic, I strongly suggest you [add ORCIDs to the authors list](https://ropensci.org/technotes/2018/10/08/orcid/)! You could also use the contributor roles here to clarify the different contributions (e.g. of Wikle)
- `pixelate`
  - The function paramter and documentation use "shapefile", which is misleading, because it takes `sf` objects as inputs, which could come from any vector file/database format. Suggest to rephrase/rename. This actually applies to the vignette, too. "Map data" would make more sense than "shapefile", right?
  - You do some projection changing in the background -  mention this in the functions docs. _What about users who want to use another projection in their visualisation?_
- I like that you have a lot of developer comments explaining why you do something in the code :+1: 
- I could install the package from source. :+1: 
- `devtools::check()` completes with one Note (good job!): you should add `paper` directory to `.Rbuildignore`

_I'd be happy to take another look at the source code, if the more general comments are resolved._ :smiley: 

### Potentially going beyond JOSS review

- Did you consider integrating with any color-scheme packages in R to help build "good" palettes?
- What does the "uv" in `read.uv` stand for? Maybe a longer function name makes it more memorable/understandable?
- `sp` is the "old way", the current way to work with vector data is the package `sf`. In the paper you write " and we will continue to improve the toolkit to enhance its utility" - it would be great if you could update your functions (or test them, automatic coercion might make a lot of things work already) to play nicely with `sf` objects.
- Is it possible to export an anmination as a GIF? Would be cool to have that in the README or vignette...

### My session info

```r
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Vizumap_1.2.0

loaded via a namespace (and not attached):
 [1] spbabel_0.5.1        httr_1.4.2           maps_3.3.0           pkgload_1.1.0        tidyr_1.1.0         
 [6] bit64_0.9-7          jsonlite_1.7.0       assertthat_0.2.1     sp_1.4-2             animation_2.6       
[11] blob_1.2.1           remotes_2.1.1        sessioninfo_1.1.1    pillar_1.4.6         RSQLite_2.2.0       
[16] backports_1.1.8      lattice_0.20-41      glue_1.4.1           digest_0.6.25        colorspace_1.4-1    
[21] plyr_1.8.6           clisymbols_1.2.0     pkgconfig_2.0.3      devtools_2.3.0       broom_0.7.0         
[26] purrr_0.3.4          scales_1.1.1         processx_3.4.3       jpeg_0.1-8.1         ggmap_3.0.0         
[31] tibble_3.0.3         farver_2.0.3         generics_0.0.2       cranlike_1.0.2       ggplot2_3.3.2       
[36] usethis_1.6.1        ellipsis_0.3.1       withr_2.2.0          cli_2.0.2            magrittr_1.5        
[41] crayon_1.3.4         memoise_1.1.0        maptools_1.0-1       ps_1.3.3             fs_1.4.2            
[46] fansi_0.4.1          crancache_0.0.0.9001 parsedate_1.2.0      foreign_0.8-79       pkgbuild_1.1.0      
[51] tools_4.0.2          prettyunits_1.1.1    RgoogleMaps_1.4.5.3  lifecycle_0.2.0      stringr_1.4.0       
[56] munsell_0.5.0        geoaxe_0.1.0         callr_3.4.3          compiler_4.0.2       rlang_0.4.7         
[61] debugme_1.1.0        grid_4.0.2           rstudioapi_0.11      rjson_0.2.20         rappdirs_0.3.1      
[66] prompt_1.0.0         labeling_0.3         bitops_1.0-6         testthat_2.3.2       gtable_0.3.0        
[71] dadjoke_0.1.2        DBI_1.1.0            curl_4.3             reshape2_1.4.4       rematch2_2.1.2      
[76] R6_2.4.1             gridExtra_2.3        rgdal_1.5-12         dplyr_1.0.0          rgeos_0.5-3         
[81] bit_1.1-15.2         rprojroot_1.3-2      desc_1.2.0           stringi_1.4.6        Rcpp_1.0.5          
[86] vctrs_0.3.2          png_0.1-7            tidyselect_1.1.0   
```

**Comment by lydialucchesi** [comment]
Hi Dr. Nüst (@nuest),

Thank you for reviewing the Vizumap paper and package. I will start
working on the revisions. Is there a date that I should have the
revisions completed by (@bstabler)?

Following is clarification regarding author contributions and licensing.

Chris and I collaborated on the development of three Vizumap
visualisation methods. Chris helped design the first three map types
and provided advice on method implementation (e.g., using the rotation
matrix to implement the glyph map). These maps are described in the
first paper referenced, which can be accessed here:
http://faculty.missouri.edu/~wiklec/LucchesiWikle2017Stat.
Accompanying that paper is a supplementary R vignette that includes
the code I wrote to produce the paper figures:
http://faculty.missouri.edu/~wiklec/sta4150-sup-0001-supplementary.

Petra and I then started a collaboration in 2017 where we translated
the methods and code from the Stat paper into an R package, and this
was housed in a CSIRO Bitbucket repository. Petra made contributions
towards speed up and design of the code to make it more user friendly
and efficient. Petra also contributed the 4th visualisation method in
the package (the exceedance probability map), which was outlined in
the second paper we reference. We are working on creating an
open-access link for this paper.

When we decided to port this code to GitHub and make it public, we had
trouble getting a CSIRO Bitbucket repository transferred to GitHub. We
therefore created a new GitHub repository that unfortunately does not
contain all of the history of commits. If it would be helpful, we can
make the original Bitbucket repository available to you as a separate
repository so you can see the version history and history of commits
to validate contributions.

The License file should not appear in that repository, and the
Description file should just include a reference to GPLv3 only. My
apologies for the error and causing confusion. I will push a change to
the repo to fix this.

Thank you for providing helpful feedback that will improve the paper
and the package.

Best regards,
Lydia

**Comment by nuest** [comment]
Just Daniel, please - no "Dr." yet :man_shrugging: 

**Thanks for the clarification!** All makes sense - for me a short comment about the project history, e.g. in the contributors documentation, or the README, would suffice to make this clear to future users/readers.

**Comment by bstabler** [comment]
@lydialucchesi - no hurry, the pace is up to you

**Comment by bstabler** [comment]
@nuest wrote "The first author @lydialucchesi has primarily contributed to the JOSS paper - at least judging from the commits. Major code contributions were made by @pkuhnert. The third author has not made any contribution to the software himself, but I assume there is a significant contribution in algorithm design and the idea behind the package. I assume the contributors discussed this, but IMO it would be best to capture this with a short comment by @pkuhnert and Christopher K. Wikle (does not seem to be on GitHub) ? @bstabler what's your take on this?"

I agree, it would be nice to document the contribution in some traceable way.

**Comment by lydialucchesi** [comment]
@nuest @bstabler great, I will make sure to add information about contributions as I complete the package revisions.

**Comment by bstabler** [comment]
@whedon remind @mikejohnson51 in 1 day

**Comment by whedon** [comment]
Reminder set for @mikejohnson51 in 1 day

**Comment by lydialucchesi** [comment]
Following is an open-access link to the second reference in the paper: https://publications.csiro.au/publications/#publication/PIcsiro:EP168206. Best, Lydia

**Comment by whedon** [comment]
:wave: @mikejohnson51, please update us on how your review is going.

**Comment by nuest** [comment]
Just FYI: I'll be on vacation next week, back Sep 15.

**Comment by lydialucchesi** [comment]
I am working on revisions and have a running list of what has been done so far saved in the paper folder in the Vizumap repo: https://github.com/lydialucchesi/Vizumap/blob/master/paper/revisionList.docx.

**Comment by bstabler** [comment]
I think @mikejohnson51 must be too bust or something so we need another reviewer.  Let me unassign him and look for another reviewer.

**Comment by bstabler** [comment]
@whedon remove @mikejohnson51 as reviewer

**Comment by whedon** [comment]
OK, @mikejohnson51 is no longer a reviewer

**Comment by nuest** [comment]
@lydialucchesi Thanks for the update!

I strongly recommend not to put docx files under version control - it is effectively a ZIP archive that git cannot effectively track changes over time with. Why not use a plain text file, with Markdown formatting?

**Comment by lydialucchesi** [comment]
@nuest Good idea. I have made the switch, and the new file can be found here: https://github.com/lydialucchesi/Vizumap/blob/master/paper/revisions.md. Thanks, Lydia

**Comment by lydialucchesi** [comment]
Hi @bstabler and @nuest, we have addressed and incorporated all of revisions from the first review (https://github.com/lydialucchesi/Vizumap/blob/master/paper/revisions.md). Please let me know if there is any additional information I can provide regarding the changes made.

Should we update the versioning from 1.0.0 to 2.0.0 given the major changes? Version 1.0.0 is from the original package release in 2017, and the name of the package at that time was VizU.

@nuest, thank you for your thoughtful, constructive feedback in the review. We are so excited with the improved state of the package and paper.

Kind regards, Lydia

**Comment by nuest** [comment]
@lydialucchesi Thanks! Will take look early next week. Could you be so kind and provide a comparison link between the version first submitted and the updated one? Thanks!

@bstabler What's your take on the version number?

I'd suggest to stick with semantic versioning and not make it a major change if there were not breaking changes.

**Comment by nuest** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02409/joss.02409/10.21105.joss.02409.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02409/joss.02409/10.21105.joss.02409.pdf) :page_facing_up: :point_left:

**Comment by lydialucchesi** [comment]
@nuest, below is a link comparing the version that was reviewed (I pushed a few minor fixes after we submitted to JOSS and prior to your review) and the updated version of Vizumap.
https://github.com/lydialucchesi/Vizumap/compare/75628e…059c21d

Here is a link to the repository prior to the revisions.
https://github.com/lydialucchesi/Vizumap/tree/75628e28b79c52355b574e699688f53a0d9f284d

**Comment by bstabler** [comment]
Hi @jsta, @jayrobwilliams, @mikerspencer, @GISerDaishaoqing - we lost one of our reviewers so I'm wondering if you would be interested in reviewing?  Thanks!

**Comment by GISerDaiShaoqing** [comment]
@bstabler Thanks for the invitation. I'm pleased to review this paper.

**Comment by bstabler** [comment]
@whedon add @GISerDaiShaoqing as reviewer

**Comment by whedon** [comment]
OK, @GISerDaiShaoqing is now a reviewer

**Comment by nuest** [comment]
@bstabler @lydialucchesi I'll do the second round of my reviews early next week. Looking forward to see what @GISerDaiShaoqing thinks! Welcome!

**Comment by nuest** [comment]
[I finally got to this again, I'm really sorry - I hope you'll get the second review soon to get this out.]

I took a look at [these changes](https://github.com/lydialucchesi/Vizumap/compare/75628e%E2%80%A6059c21d), compared them with [my comments](https://github.com/openjournals/joss-reviews/issues/2409#issuecomment-664007471), and subsequently updated [my review checklist](https://github.com/openjournals/joss-reviews/issues/2409#issue-647776407).

- It's great to see a CoC and contributor notes were added.
- License concerns are resolved and version numbers are aligned.
- Documentation comments were resolved/answered very thoroughly - good job! :rocket: 
- Authorship concerns were answered.

Quick fixes:

- Add an empty line before the pixel map plot so that the figure is not in a line with the previous sentence ending "... (2017).".
- Add a section and one-liner to the README how to run the tests, so I can tick of the final box "Automated tests: Are there automated tests or manual steps described so that the functionality of the software can be verified?". This is obvious for R experts, but it does not hurt to make it clear for any visitor to the repo. (If you're open to spend an hour on this, run the tests as a GitHub action and add a badge to the README, see link below.)

_With these very minor changes I'd be happy to see this submission published!_ Thank you @lydialucchesi !

Some further comments which from my perspective should not hold up further processing of the submission:

- I suggest to remove the `paper/paper.html` file, since it is not easily accessible form the repo and might be confusing. However, it's in the git history now anyway...
- `paper/makeFigures.R` should save the plots directly to the respective PNG files to make it perfectly clear which function creates with plot.
- I had a few minor warnings and notes when running `R CMD check` and running the tests (issues with CRSs), you might want to take a look at that, as you might consider submitting to CRAN at some point; might be worth adding a GitHub action and get started with CI, see https://ropenscilabs.github.io/actions_sandbox/

**Comment by lydialucchesi** [comment]
Hi @nuest, 

We have fixed the text under the pixel figure, added a README section about running tests, and included a badge to indicate whether the package is passing the R CMD check. 

Thank you so much for taking the time to provide thorough reviews of Vizumap! It is greatly appreciated.

Kind regards,
Lydia

**Comment by nuest** [comment]
@bstabler I'm through with my checklist. Looking forward to learn from a second reviewer what I missed.

@lydialucchesi Congrats on a solid package and article, and best of luck with the remaining review.

**Comment by GISerDaiShaoqing** [comment]
The authors present an R package for visualization of uncertainty map, Vizumap. It's a very interesting and useful tool for decision-making. The package included four main types of uncertainty map. The guideline and paper are well written under the revision. Several minor revisions should be addressed as follows. @lydialucchesi 

### **General checks**

- The 'Version' in DESCRIPTION file is still 1.0.0. May the authors clarify or revise it.

###  **Functionality**

- I also get a warning when running the *pixelate*. Although it seems successful, fixing it may be better?

`pix <- pixelate(ca_geo, id = "region")`

> Warning messages:
In wkt(obj) : CRS object has no comment
In wkt(obj) : CRS object has no comment
In gBuffer(geoData, byid = TRUE, width = 0) :
Spatial object is not projected; GEOS expects planar coordinates

### **Documentation**

- I think It's a good guidline for user to use the package now.
- My suggestion is providing a website guidline for this package using *pkgdown* or converting the page of *vignette* into html version.

### **Software paper**

- All the figures included in this paper lack caption. The authors should add the caption below each figure.
- The formulas should be numbered.
- In the section of 'Pixel map', the figure should be placed below the text of the paragraph.

![image](https://user-images.githubusercontent.com/26642320/100673435-bb2d0200-3363-11eb-93e4-2a303337d7bd.png)

- On page 2, the citations of *pixelate* and *biscale* seem different from other citations. Please use a uniform formatted citation style. For example, the name of the first author et al. (published year).

- Besides, each figure of the first, second, and third methods, Bivariate map, Pixel map, and Glyph map, actually consists of two sub-figures, the authors should use (a) and (b) to distinguish them.

**Comment by lydialucchesi** [comment]
Dear @GISerDaiShaoqing ,

Thank you for taking the time to review Vizumap. We have addressed the suggested revisions, and below are comments regarding each revision. Additionally, all edits to the package and paper have been documented here: https://github.com/lydialucchesi/Vizumap/blob/master/paper/revisions.md.

**General checks**

We accidentally submitted the wrong version number in the JOSS submission form. The correct version was 1.0.0. However, given all of the recent changes, as of today we have updated the version to 1.1.0. The description file has been updated accordingly to 1.1.0.

**Functionality**

To deal with the pixelate warnings, we now remove the projection in order to “gBuffer” and “geoaxe::chop” and then return the spatial object to its original projection. A comment regarding this has been added to the pixelate function documentation so users are aware.

**Documentation**

A pkgdown site has been generated, and a link to the site (https://lydialucchesi.github.io/Vizumap/) is provided on the README page.

We have also added a new section to the README called “History of Vizumap.” We felt this would be a good way to document how the collaboration came about and the different author contributions requested earlier in the JOSS review process.

**Software paper**

Captions have been added for each figure, and formulas have been numbered. The pixel map is now below the pixel paragraph.

When including in-text citations, we followed the JOSS guidelines provided here: https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography. The in-text citation format we chose depends on the sentence structure, and both formats are supported by JOSS. Please let us know if you would like us to reword the sentences.

We have updated the pixel map to include sub-figure labels. If possible, we would prefer to not add sub-figure labels to the bivariate and glyph maps as we would like the map and map key to be thought of as just one figure and not two separate figures.

We greatly appreciate your thoughtful review of Vizumap and are excited about the changes/additions, especially the new pkgdown site! Please let us know if there is any additional information we can provide regarding the revisions for this second JOSS review.

Kind regards,
Lydia

**Comment by lydialucchesi** [comment]
Following is a link comparing the repo that was reviewed and the updated version: https://github.com/lydialucchesi/Vizumap/compare/8b28d0f...1b96771.

**Comment by GISerDaiShaoqing** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02409/joss.02409/10.21105.joss.02409.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02409/joss.02409/10.21105.joss.02409.pdf) :page_facing_up: :point_left:

**Comment by GISerDaiShaoqing** [comment]
Dear @lydialucchesi

Sorry to reply you late. Thanks for your revision. The authors have have done a thorough job revising the manuscript accordingly. I am satisfied with the changes made. @bstabler

Congrats on such a solid package and article.

**Comment by lydialucchesi** [comment]
Thank you for reviewing Vizumap, @GISerDaiShaoqing.

We reviewed the proofs and made a few small edits.

- Capitalised all instances of "figure"
- Changed Sub-figure B to Figure 2(B) (likewise for Figure 2(A))
- Added "for the second author" in acknowledgements
- Changed "uncertainty" to "uncertainties" in two places

@bstabler, what are the next steps in the review process?

King regards,
Lydia

**Comment by arfon** [comment]
@lydialucchesi - At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by arfon** [comment]
Friendly reminder we're waiting on you to take the above next steps here @lydialucchesi before we can proceed.

**Comment by lydialucchesi** [comment]
Dear @arfon,

Thank you for your message. We have created a new release of Vizumap (v1.2.0) that includes the changes resulting from the JOSS review. We have also made an archive of the software on Zenodo. We first archived the R package on Zenodo in 2017. The latest archive for JOSS is linked to the original archive. Will that be okay?

A link to the Zenodo archive of Vizumap: 
https://doi.org/10.5281/zenodo.4554558

The DOI: 
10.5281/zenodo.4554558

Also, is it possible to get a copy of the most recent paper proof? We have made several small changes since the last proof was generated.

Kind regards,
Lydia

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.4554558 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4554558" target="_blank">10.5281/zenodo.4554558</a> is the archive.

**Comment by arfon** [comment]
@whedon set v1.2.0 as version

**Comment by whedon** [comment]
OK. v1.2.0 is the version.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/sta4.150 is OK
- 10.1071/MF17237 is OK
- 10.1214/16-AOAS950 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2105

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2105, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@lydialucchesi - this is the [latest proof](https://github.com/openjournals/joss-papers/blob/b271ac47cacfb57db85d04dd080332f41744ab96/joss.02409/10.21105.joss.02409.pdf) based on all the changes to date.

**Comment by lydialucchesi** [comment]
Thank you, @arfon. We have reviewed the latest proof and are happy with it. Best, Lydia

**Comment by danielskatz** [comment]
As far as I can tell, this is ready to publish but got dropped by us (JOSS) somehow, so I'm going to go ahead.

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1367851587901812744) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2136
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02409](https://doi.org/10.21105/joss.02409)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @lydialucchesi (Lydia Lucchesi) and co-authors!!

And thanks to @nuest and @GISerDaiShaoqing for reviewing, and @bstabler for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02409/status.svg)](https://doi.org/10.21105/joss.02409)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02409">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02409/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02409/status.svg
   :target: https://doi.org/10.21105/joss.02409
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02409/status.svg)](https://doi.org/10.21105/joss.02409)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
